### PR TITLE
Add docker container for C++ format tools

### PR DESCRIPTION
**Pull request by bogdandrutu**
To run the tools use:
`docker run --rm --privileged=true --volume ${PWD}:/otel cpp_format_tools:latest`

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: .github/workflows/cpp_format_tools.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  push:
+    tags: [ '**' ]
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+    paths:
+      - .github/workflows/cpp_format_tools.yml
+      - cpp_format_tools/Dockerfile
+      - cpp_format_tools/format.sh
```
Comment (yurishkuro) [inline .github/workflows/cpp_format_tools.yml:11]: not `cpp_format_tools/*` ?

File: .github/workflows/cpp_format_tools.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build the Docker image
+      run: docker build cpp_format_tools/. -t cpp_format_tools
+    - name: Push the Docker image
+      if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/')
+      run: |
+        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
```
Comment (yurishkuro) [inline .github/workflows/cpp_format_tools.yml:23]: may want to use `docker/login-action@v1`

File: .github/workflows/cpp_format_tools.yml (hunk 1 above)
Comment (bogdandrutu) [reply .github/workflows/cpp_format_tools.yml]: Done.

File: .github/workflows/cpp_format_tools.yml (hunk 2 above)
Comment (bogdandrutu) [reply .github/workflows/cpp_format_tools.yml]: Done.

File: cpp_format_tools/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+ARG ALPINE_VERSION=3.13
+
+FROM alpine:${ALPINE_VERSION}
+LABEL maintainer="The OpenTelemetry Authors"
+RUN apk update
+RUN apk add --no-cache clang=10.0.1-r0 python3 py3-pip git curl
```
Comment (ThomsonTan) [inline cpp_format_tools/Dockerfile:6]: Move all the version definitions to the beginning of this file, like `ALPINE_VERSION`?

Comment (bogdandrutu) [reply cpp_format_tools/Dockerfile]: Done.

**Review by ThomsonTan — APPROVED** [review]

**Comment by bogdandrutu** [comment]
@open-telemetry/technical-committee please review :)

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**